src/App.tsx

```diff
@@ -9,39 +9,29 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
-
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import * as postsActions from './features/Slices/postSlise';
+import { clearPost } from './features/Slices/selectedPostSlice';
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const { items, loaded, hasError } = useAppSelector(state => state.posts);
 
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const author = useAppSelector(state => state.author.author);
+  const selectedPost = useAppSelector(state => state.selectedPost.selectedPost);
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(clearPost());
 
     if (author) {
+      const loadUserPosts = async (userId: number) => {
+        await dispatch(postsActions.fetchPosts(userId));
+      };
+
       loadUserPosts(author.id);
     } else {
-      setPosts([]);
+      dispatch(postsActions.clearPosts());
     }
```

Comment:
Consider moving the function `loadUserPosts` outside of the `useEffect` hook. This will improve the readability of your code and make the `useEffect` hook's dependencies array cleaner. You can define it above the `useEffect` and then just call it inside.